### PR TITLE
Shadow map update rate control

### PR DESCRIPTION
**Pull request by guycalledfrank**
Not every shadow map has to be updated each frame. Some can be updated less often and some shouldn’t even update unless shadow-casting objects or light move, or other game event happens. New API:
- light.shadowUpdateMode
- light.updateShadow()

Mode can be:
- pc.SHADOW_UPDATE_NONE (don't update shadow map)
- pc.SHADOW_UPDATE_THISFRAME (this is set when you call updateShadow() - refreshes shadows at this frame only)
- pc.SHADOW_UPDATE_REALTIME (this is the default value/behaviour)

Note, that API is only for scene_light, it's not exposed in light component, and probably shouldn't.

test scene:
http://playcanvas.com/designer/339455/scene/356052

### CODE REVIEW

**Comment by willeastcott** [comment]
Couple of comments:
* pc.Light is not public API. If developers need this, it must be in the light component.
* It should be pc.SHADOWUPDATE_XXX, not pc.SHADOW_UPDATE_XXX. Check out the format of all the other enums.

**Comment by guycalledfrank** [comment]
Not sure if this should be in component, because:
- it doesn't have to be serialized
- it doesn't have to be in UI
- I'm tired of copy-pasting :) Let there be only one file containing light properties already.

**Comment by willeastcott** [comment]
There's no requirements for component properties to be serialized or have UI.
I sympathize about your tiredness, but it's the only way developers will find the API in the docs. ;)

**Comment by willeastcott** [comment]
It probably makes sense that we merge some of the scene API into the components so we don't need to do all this copy-pasting any more. But that's a separate job, right?

File: src/scene/scene_light.js [hunk 1]
```diff
@@ -443,6 +445,12 @@ pc.extend(pc, function () {
          */
         setType: function (type) {
             this._type = type;
+        },
+
+        updateShadow: function() {
+            if (this.shadowUpdateMode!==pc.SHADOWUPDATE_REALTIME) {
+                this.shadowUpdateMode = pc.SHADOWUPDATE_THISFRAME;
+            }
```
Comment (willeastcott) [inline src/scene/scene_light.js]: Doesn't this function need to go on the pc.LightComponent?

**Pull request merged**